### PR TITLE
[ACM-4128]: Deploying AWS private clusters

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-4128

This PR ports the upstream HyperShift docs about deploying AWS private clusters to the product docs.

### CODE REVIEW

**Comment by lahinson** [comment]
/assign @rokej 
/assign @enxebre 

@rokej @enxebre When you can, PTAL at this PR, which ports the upstream docs about deploying AWS private clusters to the product docs.

File: clusters/hosted_control_planes/managing_hosted_aws.adoc [hunk 1]
```diff
@@ -206,6 +207,191 @@ The access secrets for hosted control plane clusters are stored in the `hypershi
...
+[#prerequisites-aws-private-clusters]
+=== Prerequisites
+
+To enable private hosted clusters for AWS, you must first configure hosted control planes. For more information, see xref:../../clusters/hosted_control_planes/configure_hosted_aws.adoc#hosting-service-cluster-configure-aws[Configuring hosted control planes (Technology Preview)].
+
+[#create-aws-private-hosted-cluster]
+=== Creating a private hosted cluster
+
+. Create the private cluster IAM policy document by taking one of the following steps:
+** If you use shell, enter the following commands:
```
Comment (rokej) [inline clusters/hosted_control_planes/managing_hosted_aws.adoc]: Can lines 227 ~ 272 be simplified by saying...

Create the private cluster IAM policy document by entering the following command

cat << EOF >> policy.json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "ec2:CreateVpcEndpointServiceConfiguration",
        "ec2:DescribeVpcEndpointServiceConfigurations",
        "ec2:DeleteVpcEndpointServiceConfigurations",
        "ec2:DescribeVpcEndpointServicePermissions",
        "ec2:ModifyVpcEndpointServicePermissions",
        "ec2:CreateTags",
        "elasticloadbalancing:DescribeLoadBalancers"
      ],
      "Resource": "\*"
    }
  ]
}

Comment (rokej) [reply clusters/hosted_control_planes/managing_hosted_aws.adoc]: To me, the two options are the same and the next command refers to `policy.json` file. So instruct users to create the `policy.json` file.

File: clusters/hosted_control_planes/managing_hosted_aws.adoc [hunk 2]
```diff
@@ -206,6 +207,191 @@ The access secrets for hosted control plane clusters are stored in the `hypershi
...
+== Deploying AWS private clusters
+
+By default, hosted control plane guest clusters are publicly accessible through public DNS and the default router for the management cluster.
+
+For private clusters on AWS, all communication with the guest cluster occurs over AWS PrivateLink. To configure hosted control planes for private cluster support on AWS, take the following steps.
+
+*Important:* Although public clusters can be created in any region, private clusters can be created only in the region that is specified by `--aws-private-region`.
+
+[#prerequisites-aws-private-clusters]
+=== Prerequisites
```
Comment (rokej) [inline clusters/hosted_control_planes/managing_hosted_aws.adoc:120]: https://github.com/stolostron/rhacm-docs/blob/lahinson-acm-4128-aws-private-clusters/clusters/hosted_control_planes/configure_hosted_aws.adoc#enabling-aws-privatelink is a pre-req.

File: clusters/hosted_control_planes/managing_hosted_aws.adoc [hunk 3]
```diff
@@ -206,6 +207,191 @@ The access secrets for hosted control plane clusters are stored in the `hypershi
...
+aws iam attach-user-policy --user-name=hypershift-operator --policy-arn=$POLICY_ARN
+----
+
+. Create an IAM access key for the user by entering this command:
++
+----
+aws iam create-access-key --user-name=hypershift-operator
+----
+
+. Create a credentials file, `$AWS_PRIVATE_CREDS`, with the access ID and key for the user that you created, by entering these commands:
```
Comment (rokej) [inline clusters/hosted_control_planes/managing_hosted_aws.adoc]: Remove lines 298 ~ 323 about installing the hypershift operator. It should already be covered by the pre-req I mentioned above https://github.com/stolostron/rhacm-docs/blob/lahinson-acm-4128-aws-private-clusters/clusters/hosted_control_planes/configure_hosted_aws.adoc#enabling-aws-privatelink

File: clusters/hosted_control_planes/managing_hosted_aws.adoc (hunk 1 above)
Comment (lahinson) [reply clusters/hosted_control_planes/managing_hosted_aws.adoc]: Works for me

File: clusters/hosted_control_planes/managing_hosted_aws.adoc (hunk 2 above)
Comment (lahinson) [reply clusters/hosted_control_planes/managing_hosted_aws.adoc:120]: Done

File: clusters/hosted_control_planes/managing_hosted_aws.adoc (hunk 3 above)
Comment (lahinson) [reply clusters/hosted_control_planes/managing_hosted_aws.adoc]: Done

**Comment by lahinson** [comment]
@rokej Thanks for your review! I incorporated your feedback. Let me know if anything else should change.

File: clusters/hosted_control_planes/managing_hosted_aws.adoc [hunk 4]
```diff
@@ -206,6 +207,141 @@ The access secrets for hosted control plane clusters are stored in the `hypershi
 - `kubeconfig` secret: `<hostingNamespace>-<name>-admin-kubeconfig` (clusters-hypershift-demo-admin-kubeconfig)
 - `kubeadmin` password secret: `<hostingNamespace>-<name>-kubeadmin-password` (clusters-hypershift-demo-kubeadmin-password)
 
+[#deploying-aws-private-clusters]
+== Deploying AWS private clusters
```
Comment (rokej) [inline clusters/hosted_control_planes/managing_hosted_aws.adoc]: There is another section called `Deploying a hosted cluster on AWS`. This `Deploying AWS private clusters` is an alternative way of `Deploying a hosted cluster on AWS`. So I am wondering if we should re-organize these a little bit. 

First, I think we should rename `Deploying AWS private clusters` to `Deploying a private hosted cluster on AWS` then put this section right under `Deploying a hosted cluster on AWS`. 

We should clearly indicate that a user follows either `Deploying a hosted cluster on AWS` or `Deploying a private hosted cluster on AWS`.

File: clusters/hosted_control_planes/managing_hosted_aws.adoc [hunk 5]
```diff
@@ -206,6 +207,141 @@ The access secrets for hosted control plane clusters are stored in the `hypershi
...
+----
+<1> The `--endpoint-access` flag designates whether a cluster is public or private.
+
+The API endpoints for the cluster are accessible through a private DNS zone:
+
+- `api.$CLUSTER_NAME.hypershift.local`
+- `*.apps.$CLUSTER_NAME.hypershift.local`
+
+[#access-aws-private-hosted-cluster]
+=== Accessing a private hosted cluster
```
Comment (rokej) [inline clusters/hosted_control_planes/managing_hosted_aws.adoc]: Similar to my previous comment on `Deploying a hosted cluster on AWS` section and `Deploying a private hosted cluster on AWS`, there is an existing section called `Accessing a hosting cluster on AWS`. So if user followed `Deploying a hosted cluster on AWS` section, then the user should follow `Accessing a hosting cluster on AWS`. If user followed `Deploying a private hosted cluster on AWS` to create a private cluster, then the user should follow `Accessing a private hosted cluster` section.

I also think `Accessing a private hosted cluster` should be renamed to `Accessing a private hosting cluster on AWS`.

**Comment by lahinson** [comment]
@rokej Thanks again for your feedback! I agree that the content would make more sense if it were organized differently. I reorganized the sections according to your comments. PTAL when you can and let me know what you think.

**Review by rokej — APPROVED** [review]
Thank you Laura.

File: clusters/hosted_control_planes/managing_hosted_aws.adoc [hunk 6]
```diff
@@ -92,12 +97,157 @@ hypershift create cluster aws \
...
+----
+ssh -o ProxyCommand="ssh ec2-user@$BASTION_IP -W %h:%p" core@$NODE_IP
+----
+
+. From the SSH shell, copy the `kubeconfig` file contents to a file on the node by entering the following commands:
++
+----
+cat << EOF >> kubeconfig
+<paste kubeconfig contents>
+EOF
```
Comment (swopebe) [inline clusters/hosted_control_planes/managing_hosted_aws.adoc]: We don't usually have EOF or `# ...` in our YAML samples, at least we try to remove them in editing.

**Review by swopebe — APPROVED** [review]
one comment, thanks!

File: clusters/hosted_control_planes/managing_hosted_aws.adoc (hunk 6 above)
Comment (lahinson) [reply clusters/hosted_control_planes/managing_hosted_aws.adoc]: Fixed. Good catch.

**Pull request merged**